tests/deploy_test.go

```diff
@@ -1,18 +1,94 @@
 package tests
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"net/http"
 	"os"
 	"strings"
 	"testing"
 
+	"github.com/openfaas/faas-provider/types"
 	"github.com/openfaas/faas/gateway/requests"
 )
 
 var emptyQueryString = ""
 
+func Test_Scaling_Up(t *testing.T) {
+	functionRequest := requests.CreateFunctionRequest{
+		Image:      "functions/alpine:latest",
+		Service:    "test-scaling",
+		Network:    "func_functions",
+		EnvProcess: "env",
+	}
+
+	deployStatus, deployErr := deploy(t, functionRequest)
+	if deployErr != nil {
+		t.Errorf(deployErr.Error())
+		t.Fail()
+		return
+	}
+
+	if deployStatus != http.StatusOK && deployStatus != http.StatusAccepted {
+		t.Errorf("got %d, but want %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
+	}
+
+	invoke(t, functionRequest.Service, emptyQueryString, http.StatusOK)
+
+	var want uint64 = 2
+	scaleStatus, err := scale(functionRequest.Service, want)
+	if err != nil {
+		t.Error(err)
+	}
+	if scaleStatus != http.StatusOK && scaleStatus != http.StatusAccepted {
+		t.Errorf("got %d, but want %d or %d", scaleStatus, http.StatusOK, http.StatusAccepted)
+	}
+
+	replicas, getReplicasStatus, err := getReplicas(functionRequest.Service)
+	if err != nil {
+		t.Error(err)
+	}
+	if getReplicasStatus != http.StatusOK && getReplicasStatus != http.StatusAccepted {
+		t.Errorf("got %d, but want %d or %d", getReplicasStatus, http.StatusOK, http.StatusAccepted)
+	}
+
+	if want != replicas {
+		t.Errorf("got %d, but want %d replicas", replicas, want)
+	}
+}
+
+func scale(fnName string, replicas uint64) (int, error) {
+	req := types.ScaleServiceRequest{
+		ServiceName: fnName,
+		Replicas:    replicas,
+	}
+
+	reqBody := new(bytes.Buffer)
+	json.NewEncoder(reqBody).Encode(req)
+
+	endpoint := os.Getenv("gateway_url") + "system/scale-function/" + fnName
```

Comment:
Can you use the new approach to parse a URL then set the path? Have a look at master.

Reply:
Done